### ISSUE TITLE
Make improvements to documentation page on performance tips

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1893]
In #1887, we added a page on performance tips for using PlasmaPy (see also #1760).  This PR is to keep track of remaining improvements that we could do as time permits.  

 - [ ] Expand the discussion on lite-functions
 - [ ] Use `timeit.timeit` to show actual timings and to make sure we're not making this all up
 - [ ] Say a bit more about when is a good time to optimize performance vs. when it isn't (maybe adding the quote about how premature optimization?)

**Comment by StanczakDominik** [comment]
`timeit` is nice, but I always just do these in Ipython with `%timeit`.

**Comment by namurphy** [comment]
Agreed about `%timeit`!  I was having some trouble getting `timeit` to work in regular Python.  Given your suggestion, I'm wondering if this page would be best converted into a Jupyter notebook.
